src/main.rs

```diff
@@ -86,3 +86,45 @@ fn print_verbose_err() {
          https://github.com/rust-lang/rust-bindgen/issues/new"
     );
 }
+
+#[cfg(test)]
+mod test {
+    use std::path::PathBuf;
+
+    fn build_flags_output_helper(builder: &bindgen::Builder) {
+        let mut command_line_flags = builder.command_line_flags();
+        command_line_flags.insert(0, "bindgen".to_string());
+
+        let flags_quoted: Vec<String> = command_line_flags
+            .iter()
+            .map(|x| format!("'{}'", x))
+            .collect();
+        let flags_str = flags_quoted.join(" ");
+        println!("{}", flags_str);
+
+        crate::options::builder_from_flags(command_line_flags.into_iter())
+            .unwrap();
+    }
+
+    #[test]
+    fn commandline_flag_roundtrip() {
+        // test1: various options
+        let bindings = bindgen::Builder::default()
+            .header("tests/headers/char.h")
+            .record_matches(false)
+            .size_t_is_usize(true)
+            .rustfmt_bindings(false)
+            .rustfmt_configuration_file(Some(PathBuf::from("/dev/null")))
+            .no_partialeq(".")
+            .no_copy(".")
+            .no_hash(".");
+        build_flags_output_helper(&bindings);
+
+        // test2: multiple headers
+        let bindings = bindgen::Builder::default()
```

Comment:
We don't have a way to test this one, so this is still worth testing if you want :)

Reply:
:smile: This test is only here so that we can run run tests test with `--nocapture` to get the command output to test manually.